### ISSUE TITLE
Is polling for cursor status overkill?

### ISSUE THREAD

**Issue by TyOverby** [PistonDevelopers/piston#493]
While simply filling out a function defined on the Game trait is great for methods like render() and update(), I wonder if adapting this for cursor capture is overkill.  Cursor modes rarely change inside of a game, so instead of calling should_capture_cursor() every frame, I think leaving it as a defined method on Game that implementors could call would be best.

Instead of 

``` rust
impl Game for App {
  fn should_capture_cursor(&mut self) {
    return self.in_menu || self.player_died;
  }
}
```

you would have 

``` rust
impl Game for App {
  fn update(......) {
    ......
    self.capture_cursor(true);
    ......
    self.capture_cursor(false);
  }
}
```

Basically I worry that in order to do anything complex, a user will have to export global state in order to have their `should_capture_cursor` function respond accordingly.
